MatrixSDK/MXSession.m

```diff
@@ -615,24 +619,31 @@ - (void)handleSyncResponse:(MXSyncResponse *)syncResponse
         // Update live event stream token
         NSLog(@"[MXSession] Next sync token: %@", syncResponse.nextBatch);
         self.store.eventStreamToken = syncResponse.nextBatch;
+        
+        //  complete completion block to be called at the end of the process
+        void (^completionBlock)(void) = ^{
+            if (completion)
+            {
+                completion();
+            }
+            
+            // Broadcast that a server sync has been processed.
+            [[NSNotificationCenter defaultCenter] postNotificationName:kMXSessionDidSyncNotification
+                                                                object:self
+                                                              userInfo:@{
+                                                                  kMXSessionNotificationSyncResponseKey: syncResponse
+                                                              }];
+        };
 
-        // Commit store changes done in [room handleMessages]
-        if ([self.store respondsToSelector:@selector(commit)])
+        // Commit store changes
+        if ([self.store respondsToSelector:@selector(commitWithCompletion:)])
         {
-            [self.store commit];
+            [self.store commitWithCompletion:completionBlock];
```

Comment:
We should continue to call `completion()` as soon as possible like before.
The data is loaded in memory. The app can work. The /sync stream can run.

If we wait for `commitWithCompletion`, we will delay the usage of the app for 70s in M's case.

Reply:
Updated.

Reply:
Updated.